src/com/spikes2212/command/genericsubsystem/LambdaSubsystem.java

```diff
@@ -0,0 +1,73 @@
+package com.spikes2212.command.genericsubsystem;
+
+import java.util.function.*;
+
+/**
+ * This class represents a generic subsystem that within a limitation or
+ * without one, using lambdas for the movement and limitation.
+ *
+ * @author Eran "Eren" Goldstein
+ */
+public class LambdaSubsystem extends GenericSubsystem {
+    public final Predicate<Double> canMove;
+    public final Consumer<Double> speedConsumer;
+    
+    /**
+     * Constructs a new instance of LambdaSubsystem with the given limitations and movement.
+     *
+     * @param canMove a {@link Predicate} that takes speed and returns whether the subsystem can move
+     * @param speedConsumer a {@link Consumer} that takes speed and applies it to the subsystem
+     */
+    public LambdaSubsystem(Predicate<Double> canMove, Consumer<Double> speedConsumer) {
+        super();
+        this.canMove = canMove;
+        this.speedConsumer = speedConsumer;
+    }
+    
+    /**
+     * Constructs a new instance of LambdaSubsystem with the given limitations, movement, minSpeed and maxSpeed.
+     *
+     * @param canMove a {@link Predicate} that takes speed and returns whether the subsystem can move
+     * @param speedConsumer a {@link Consumer} that takes speed and applies it to the subsystem
+     * @param minSpeed the minimum speed
+     * @param maxSpeed he maximum speed
+     */
+    public LambdaSubsystem(Predicate<Double> canMove, Consumer<Double> speedConsumer, double minSpeed, double maxSpeed) {
```

Comment:
please add another constructor for a case when the subsystem has no limit (shooter for example). just so we can make `LambdaSubsystem` require even less code for simpler subsystems